src/xmipp/libraries/reconstruction/resolution_fso.cpp

```diff
@@ -1400,482 +1140,148 @@ class ProgResolutionDirectionalFsc : public XmippProgram
-				double cosine = fabs(iun);
-				++n;
-
-				if (cosine>=cosAngle)
-					{
-						if (f>maxFreq)
-							continue;
-
-						int idx = (int) round(f * m1sizeX);
-						std::complex<double> &z1 = dAkij(FT1, k, i, j);
-						std::complex<double> &z2 = dAkij(FT2, k, i, j);
-						dAi(z1r,idx) += real(z1);
-						dAi(z1i,idx) += imag(z1);
-						dAi(z2r,idx) += real(z2);
-						dAi(z2i,idx) += imag(z2);
-						dAi(counter,idx) += 1.0;
-					}
+					double value = DIRECT_MULTIDIM_ELEM(threeD_FSC, n) /= DIRECT_MULTIDIM_ELEM(normalizationMap, n);
+					if (std::isnan(value) == 1)
+						value = 1.0;
+					
+					if ((DIRECT_MULTIDIM_ELEM(threeD_FSC, n)> thrs) )//&& (DIRECT_MULTIDIM_ELEM(aniFilter, n) <1))
+						DIRECT_MULTIDIM_ELEM(aniFilter, n) = 1;
+					
+					size_t idx = DIRECT_MULTIDIM_ELEM(arr2indx, n);
+					DIRECT_MULTIDIM_ELEM(d3_FSCMap, idx) = value;
+					DIRECT_MULTIDIM_ELEM(d3_aniFilter, idx) = DIRECT_MULTIDIM_ELEM(aniFilter, n);
 			}
-		}
-	}
 
-	FOR_ALL_ELEMENTS_IN_ARRAY1D(sig)
-	{
-		dAi(z1r,i) = dAi(z1r,i)/dAi(counter,i);
-		dAi(z1i,i) = dAi(z1i,i)/dAi(counter,i);
-		dAi(z2r,i) = dAi(z2r,i)/dAi(counter,i);
-		dAi(z2i,i) = dAi(z2i,i)/dAi(counter,i);
-	}
-
-	MetaData mdRes;
-	size_t id;
-	FOR_ALL_ELEMENTS_IN_ARRAY1D(freq)
-	{
-		if (i>0)
-		{
-			id=mdRes.addObject();
-			dAi(freq,i) = (float) i / (m1sizeX * sampling_rate);
-			mdRes.setValue(MDL_RESOLUTION_FREQ,dAi(freq, i),id);
-			mdRes.setValue(MDL_VOLUME_SCORE1, dAi(z1r, i),id);
-			mdRes.setValue(MDL_VOLUME_SCORE2, dAi(z1i, i),id);
-			mdRes.setValue(MDL_VOLUME_SCORE3, dAi(z2r, i),id);
-			mdRes.setValue(MDL_VOLUME_SCORE4, dAi(z2i, i),id);
-		}
-	}
-
-	FileName fnmd;
-	fnmd = fn_fscmd_folder + formatString("ssnr_%i.xmd", dire);
-	mdRes.write(fnmd);
-
-}
-
-
-void getErrorCurves(int &m1sizeX, int &m1sizeY, int &m1sizeZ,
-	 Matrix1D<double> &freq_fourier_x,
-	 Matrix1D<double> &freq_fourier_y,
-	 Matrix1D<double> &freq_fourier_z, MultidimArray<double> &freqMap, size_t Nrealization, double thrs)
-{
-	MultidimArray<double> half1, half2;
-
-	MultidimArray<double> &phalf1 = half1, &phalf2 = half2, auxhalf1, auxhalf2;
-
-	Image<double> mask;
-	MultidimArray<double> &pmask = mask();
-
-	std::cout << "Reading data..." << std::endl;
-	Image<double> imgHalf1, imgHalf2;
-	imgHalf1.read(fnhalf1);
-	imgHalf2.read(fnhalf2);
-
-//		Image<double> svImg;
-//		svImg() = half1;
-//		FileName fnmd;
-//		fnmd = fn_fscmd_folder+formatString("half1_%i.mrc", k);
-//		svImg.write()
-//		svImg() = half2;
-
-	MultidimArray<double> aniParam, fsc, freq;
-
-	FileName fnmd;
-	for (size_t k=0; k<Nrealization; k++)
-	{
-		half1 = imgHalf1();
-		half2 = imgHalf2();
-
-//		half1 = auxhalf1;
-//		half2 = auxhalf2;
-		Monogenic mono;
-		double stddev = 0.5;
-		mono.addNoise(half1, 0.0, stddev);
-		mono.addNoise(half2, 0.0, stddev);
-
-		if (! fnmask.isEmpty())
-		{
-			mask.read(fnmask);
-			FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(pmask)
+			// This code fix the empty line line in Fourier space
+			size_t auxVal;
```

Comment:
extract method
also, do you know why you have an empty line there, or are you fixing some bug originating elsewhere?

Reply:
Because when we transform to Fourier space one axis is repeated. Thats the reason of the empty line. It is and issue of the library.